src/runtime/virtcontainers/kata_agent.go

```diff
@@ -1304,11 +1306,12 @@ func (k *kataAgent) createContainer(ctx context.Context, sandbox *Sandbox, c *Co
 
 	// Block based volumes will require some adjustments in the OCI spec, and creation of
 	// storage objects to pass to the agent.
-	volumeStorages, err := k.handleBlkOCIMounts(c, ociSpec)
+	layerStorages, volumeStorages, err := k.handleBlkOCIMounts(c, ociSpec)
 	if err != nil {
 		return nil, err
 	}
 
+	ctrStorages = append(layerStorages, ctrStorages...)
```

Comment:
why are the args flipped compared to the usual convention - is the order somehow important?

Reply:
Because we want to *prepend* to `ctrStorages`. See my previous comment on why.